src/front/spv/mod.rs

```diff
@@ -2098,66 +2479,134 @@ impl<I: Iterator<Item = u32>> Parser<I> {
+                        body.data.push(BodyFragment::BlockId(block_id));
+
+                        merger(body, info);
+
+                        return Ok(());
+                    }
+
+                    block_ctx.info.entry(target_id).or_insert(body_idx);
+
+                    break None;
                 }
                 Op::BranchConditional => {
                     inst.expect_at_least(4)?;
 
-                    let condition_id = self.next()?;
-                    let condition = self.lookup_expression.lookup(condition_id)?.handle;
+                    let condition = {
+                        let condition_id = self.next()?;
+                        let lexp = self.lookup_expression.lookup(condition_id)?;
+                        get_expr_handle!(condition_id, lexp)
+                    };
 
                     let true_id = self.next()?;
                     let false_id = self.next()?;
 
-                    break Terminator::BranchConditional {
-                        condition,
-                        true_id,
-                        false_id,
+                    let accept = block_ctx.bodies.len();
+                    let mut accept_block = Body {
+                        parent: body_idx,
+                        data: Vec::new(),
+                    };
+
+                    if let Some(info) = block_ctx.mergers.get(&true_id) {
+                        merger(&mut accept_block, info)
+                    }
+
+                    block_ctx.bodies.push(accept_block);
+                    block_ctx.info.entry(true_id).or_insert(accept);
+
+                    let reject = block_ctx.bodies.len();
+                    let mut reject_block = Body {
+                        parent: body_idx,
+                        data: Vec::new(),
                     };
+
+                    if let Some(info) = block_ctx.mergers.get(&false_id) {
+                        merger(&mut reject_block, info)
+                    }
+
+                    block_ctx.bodies.push(reject_block);
+                    block_ctx.info.entry(false_id).or_insert(reject);
+
+                    block.extend(emitter.finish(expressions));
+                    block_ctx.blocks.insert(block_id, block);
+                    let body = &mut block_ctx.bodies[body_idx];
+                    // Make sure the vector has space for at least two more allocations
+                    body.data.reserve(2);
+                    body.data.push(BodyFragment::BlockId(block_id));
+                    body.data.push(BodyFragment::Conditional {
+                        condition,
+                        accept,
+                        reject,
+                    });
+
+                    return Ok(());
                 }
                 Op::Switch => {
                     inst.expect_at_least(3)?;
                     let selector = self.next()?;
                     let default_id = self.next()?;
 
+                    if let Some(merge) = merge_block {
```

Comment:
I should better document this part, essentially `merge_block` is only some if the previous instruction was a `OpSelectionMerge` this is done because a `MergeBlockInformaton::SelectionMerge` needs to be changed to a `SwitchMerge` to generate `Break`s but a `LoopMerge` shouldn't be changed otherwise `Continue`s won't be generated.

Reply:
This is now documented